src/sage/rings/semirings/tropical_variety.py

```diff
@@ -0,0 +1,272 @@
+    
+    def __init__(self, poly, equations):
+        self.poly = poly
+        self._hypersurface = []
+        dim_param = len(equations[0][0]) - 1
+        vars = [SR.var('t{}'.format(i)) for i in range(1, dim_param+1)]
+        for arg in equations:
+            if not isinstance(arg, list):
+                raise ValueError("Input must be a list")
+            else:
+                subs_dict = {}
+                index_vars = 0
+                new_eq = []
+                for eq in arg[0]:
+                    var_eq = eq.variables()
+                    for var in var_eq:
+                        if var not in subs_dict:
+                            subs_dict[var] = vars[index_vars]
+                            index_vars += 1
+                    new_eq.append(eq.subs(subs_dict))
+                new_eq = tuple(new_eq)
+                arg.remove(arg[0])
+                arg.insert(0, new_eq)
+
+                params = arg[1]
+                arg.remove(params)
+                new_param = []
+                for param in params:
+                    lhs = param.lhs().subs(subs_dict)
+                    rhs = param.rhs().subs(subs_dict)
+                    if param.operator() == operator.gt:
+                        expr = lhs >= rhs
+                    else:
+                        expr = lhs <= rhs
+                    new_param.append(expr)
+                arg.insert(1, new_param)
+
+            self._hypersurface.append(arg)   
+
+    def dimension(self):
+        return len(self._hypersurface[0][0])
+    
+    def number_of_components(self):
+        return len(self._hypersurface)
+    
+    def _repr_(self):
+        components = "\n".join([f"{row}" for row in self._hypersurface])
+        return (f"Tropical hypersurface of {self.poly} are \n[{components}]")
+
+class TropicalCurve(TropicalVariety):
+    r""""
+    Represent a tropical curve in `\mathbb{R}^2`. The representation is in 
+    the form of list of lists, where the inner list represent each line 
+    segments of tropical roots.
+
+    """
+
+    def __init__(self, poly, equations):
+        TropicalVariety.__init__(self, poly, equations)        
+    
+    def _axes(self):
+        """
+        Set the default axes for ``self``
+
+        OUTPUT: A list of two lists, where the first inner list represent
+        value of x-axis and the second inner list represent value of y-axis
+
+        """
+        xmin = xmax = list(self.vertex())[0][0]
+        for vertice in self.vertex():
+            if vertice[0] < xmin:
+                xmin = vertice[0]
+            elif vertice[0] > xmax:
+                xmax = vertice[0]
+        
+        ymin = ymax = list(self.vertex())[0][1]
+        for vertice in self.vertex():
+            if vertice[1] < ymin:
+                ymin = vertice[1]
+            elif vertice[1] > ymax:
+                ymax = vertice[1]
+        
+        return [[xmin, xmax], [ymin, ymax]]
+    
+    def vertex(self):
```

Comment:
```suggestion
    def vertices(self):
```
Plural form should be used.